core/src/main/java/com/linecorp/armeria/client/BlockingWebClient.java

```diff
@@ -0,0 +1,417 @@
+     */
+    static BlockingWebClient of(URI uri) {
+        return WebClient.of(uri).blocking();
+    }
+
+    /**
+     * Sends the specified HTTP request.
+     */
+    default AggregatedHttpResponse execute(HttpRequest req) {
+        return execute(req, RequestOptions.of());
+    }
+
+    /**
+     * Sends the specified HTTP request with the specified {@link RequestOptions}.
+     */
+    AggregatedHttpResponse execute(HttpRequest req, RequestOptions options);
+
+    /**
+     * Sends the specified HTTP request.
+     */
+    default AggregatedHttpResponse execute(AggregatedHttpRequest aggregatedReq) {
+        requireNonNull(aggregatedReq, "aggregatedReq");
+        return execute(aggregatedReq.toHttpRequest());
+    }
+
+    /**
+     * Sends an empty HTTP request with the specified headers.
+     */
+    default AggregatedHttpResponse execute(RequestHeaders headers) {
+        return execute(HttpRequest.of(headers));
+    }
+
+    /**
+     * Sends an HTTP request with the specified headers and content.
+     */
+    default AggregatedHttpResponse execute(RequestHeaders headers, HttpData content) {
+        return execute(HttpRequest.of(headers, content));
+    }
+
+    /**
+     * Sends an HTTP request with the specified headers and content.
+     */
+    default AggregatedHttpResponse execute(RequestHeaders headers, byte[] content) {
+        return execute(HttpRequest.of(headers, HttpData.wrap(content)));
+    }
+
+    /**
+     * Sends an HTTP request with the specified headers and content.
+     */
+    default AggregatedHttpResponse execute(RequestHeaders headers, String content) {
+        return execute(HttpRequest.of(headers, HttpData.ofUtf8(content)));
+    }
+
+    /**
+     * Sends an HTTP request with the specified headers and content.
+     */
+    default AggregatedHttpResponse execute(RequestHeaders headers, String content, Charset charset) {
+        return execute(HttpRequest.of(headers, HttpData.of(charset, content)));
+    }
+
+    /**
+     * Prepares to send an {@link HttpRequest} using fluent builder.
+     * <pre>{@code
+     * WebClient webClient = WebClient.of(...);
+     * BlockingWebClient client = webClient.blocking();
+     * AggregatedHttpResponse response = client.prepare()
+     *                                         .post("/foo")
+     *                                         .header(HttpHeaderNames.AUTHORIZATION, ...)
+     *                                         .content(MediaType.JSON, ...)
+     *                                         .execute();
+     * }</pre>
+     */
+    BlockingWebClientRequestPreparation prepare();
+
+    /**
+     * Sends an HTTP OPTIONS request.
+     */
+    default AggregatedHttpResponse options(String path) {
+        return options(path, null);
+    }
+
+    /**
+     * Sends an HTTP OPTIONS request, appending the given query parameters to the path.
+     */
+    default AggregatedHttpResponse options(String path, @Nullable QueryParams params) {
```

Comment:
I realized `WebClient` accepts a non-null `QueryParams`. Should we update `WebClient` as well to accept `null`? Also, what do you think about making `content` nullable as well? (Ditto for `WebClient`)

Reply:
> Should we update WebClient as well to accept null?

If we allow setting `null`, users need to specify its type when `null` is set.
<img width="716" alt="image" src="https://user-images.githubusercontent.com/1866157/150486277-32031642-9fa4-4eb8-a0bc-7af97f72e2e0.png">
How about adding variants that only take `path` like `AggregatedHttpResponse post(String path)`?

Reply:
> I realized WebClient accepts a non-null QueryParams. Should we update WebClient as well to accept null?

Good suggestion.

Reply:
> If we allow setting null, users need to specify its type when null is set.

That's a good point. Let's keep `content` non-null for now, then.